From d239357f89c9eca3159e54f5daca4b6ead98402b Mon Sep 17 00:00:00 2001
From: Leo <13928879+leo9800@users.noreply.github.com>
Date: Thu, 1 May 2025 11:44:53 +0800
Subject: [PATCH] pldmlib/pldm_utils.cpp: fix missing header

addressing issue:

```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I. -I../common -Wall -W -g -MP -MD -pipe -fPIC -DPLDMLIB_EXPORT -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -DHAVE_TERMIOS_H -DHAVE_SYS_PCI_H -DNO_RDMEM -DMST_UL -MT pldm_utils.lo -MD -MP -MF .deps/pldm_utils.Tpo -c pldm_utils.cpp  -fPIC -DPIC -o .libs/pldm_utils.o
In file included from pldm_utils.cpp:21:
pldm_utils.h:39:6: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
   39 | enum class DescriptorType : uint16_t
      | ~~~~ ^~~~~
      |      -----
pldm_utils.h:39:27: error: found ':' in nested-name-specifier, expected '::'
   39 | enum class DescriptorType : uint16_t
      |                           ^
      |                           ::
pldm_utils.h:39:12: error: 'DescriptorType' has not been declared
   39 | enum class DescriptorType : uint16_t
      |            ^~~~~~~~~~~~~~
pldm_utils.h:40:1: error: expected unqualified-id before '{' token
   40 | {
      | ^
pldm_utils.cpp:32:64: error: 'uint16_t' was not declared in this scope
   32 | const std::vector<std::tuple<std::string, ComponentIdentifier, uint16_t, uint16_t, uint16_t>> components_list = {
      |                                                                ^~~~~~~~
pldm_utils.cpp:22:1: note: 'uint16_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
   21 | #include "pldm_utils.h"
  +++ |+#include <cstdint>
   22 |
pldm_utils.cpp:32:74: error: 'uint16_t' was not declared in this scope
   32 | const std::vector<std::tuple<std::string, ComponentIdentifier, uint16_t, uint16_t, uint16_t>> components_list = {
      |                                                                          ^~~~~~~~
pldm_utils.cpp:32:74: note: 'uint16_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
pldm_utils.cpp:32:84: error: 'uint16_t' was not declared in this scope
   32 | const std::vector<std::tuple<std::string, ComponentIdentifier, uint16_t, uint16_t, uint16_t>> components_list = {
      |                                                                                    ^~~~~~~~
pldm_utils.cpp:32:84: note: 'uint16_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
pldm_utils.cpp:32:84: error: template argument 3 is invalid
pldm_utils.cpp:32:84: error: template argument 4 is invalid
pldm_utils.cpp:32:84: error: template argument 5 is invalid
pldm_utils.cpp:32:92: error: template argument 1 is invalid
   32 | const std::vector<std::tuple<std::string, ComponentIdentifier, uint16_t, uint16_t, uint16_t>> components_list = {
      |                                                                                            ^~
pldm_utils.cpp:32:92: error: template argument 2 is invalid
pldm_utils.cpp:32:95: error: scalar object 'components_list' requires one element in initializer
   32 | const std::vector<std::tuple<std::string, ComponentIdentifier, uint16_t, uint16_t, uint16_t>> components_list = {
      |                                                                                               ^~~~~~~~~~~~~~~
make[2]: *** [Makefile:504: pldm_utils.lo] Error 1
make[2]: Leaving directory '/home/leo/projects/mstflint/pldmlib'
make[1]: *** [Makefile:576: all-recursive] Error 1
make[1]: Leaving directory '/home/leo/projects/mstflint'
make: *** [Makefile:461: all] Error 2
```
---
 pldmlib/pldm_utils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pldmlib/pldm_utils.cpp b/pldmlib/pldm_utils.cpp
index dfae375d..0c3a777e 100644
--- a/pldmlib/pldm_utils.cpp
+++ b/pldmlib/pldm_utils.cpp
@@ -13,6 +13,7 @@
 #include <sys/time.h>
 #include <algorithm>
 #include <cstdarg>
+#include <cstdint>
 #include <cstdio>
 #include <fstream>
 #include <sstream>
